### DISCUSSION TITLE
Image Flush Taking LONG time.

### DISCUSSION THREAD

**Discussion by Tonycaf1** [aaronwmorris/indi-allsky#1493] [Q&A]
Hello Aaron. I noticed that flushing images through the web interface can sometimes take as long as 15 or 20 minutes. Sometimes I will have to do this several times. Not a big deal but was wondering if this is unusual behavior. Running on an Ripi4. If you need any logs let me know thanks! LOVE your software!

**Comment by aaronwmorris** [comment]
Thank you!  Unfortunately, this is not unusual behavior.  Deleting files is slow because the following actions all have to happen for every file:

1. Delete thumbnail image file
2. Delete thumbnail database entry
3. Commit transaction
4. Delete image file
5. Delete image database entry
6. Commit transaction

There are probably better ways of going about this, but this is the process I landed on.  I could probably write a command line script to perform this, that way the process cannot timeout.

**Comment by aaronwmorris** [comment]
Merged #1506 with a CLI script for flushing images and videos.

```
./misc/flush_images.py
```
